### PR TITLE
compare namespace in counter and min_count

### PR DESCRIPTION
**Pull request by nadgeri14**
fix for [#2292](https://github.com/allenai/allennlp/issues/2292)

### CODE REVIEW

**Comment by nadgeri14** [comment]
@matt-gardner Can you please review the PR?
Thanks :)

**Comment by matt-gardner** [comment]
Two issues:
1. There are some failing tests - can you look at the [build log](http://build.allennlp.org/viewLog.html?buildId=24433&buildTypeId=AllenNLP_AllenNLPPullRequests&tab=buildLog) and figure out what tests are failing, and why, and fix them?
2. Can you add a test to `vocabulary_test.py` that will make sure this implementation is correct? The logic is simple enough, but a test like that makes sure that we don't have regressions later.

**Comment by nadgeri14** [comment]
@matt-gardner The build error is while executing:
`Unit Tests (pytest)` in `allennlp/common/testing/model_test_case.py:33`

If I am not mistaken for example if we have

`counter = defaultdict(<function Vocabulary.from_instances.<locals>.<lambda> at 0x7f3cd59e0510>, {'source_tokens': defaultdict(<c...: 1, 'secret': 1, 'sneaky': 1, 'retain': 1, 'composure': 1, 'ecstatic': 1, 'not': 1, 'disruption': 1, 'bursting': 1})})`

`min_count = {'tokens': 1}`
It should throw an error saying **tokens** is not present in **counter**, is my assmuption right?

**Comment by matt-gardner** [comment]
Yes, that's what's happening.  You need to figure out what's wrong in the code that's causing that error, then fix it, and put the fix in this PR.  Probably easiest if you run the failing test locally.

**Comment by brendan-ai2** [comment]
@nadgeri14, it would be great to have this fix in if you still have the time to work on it!

**Comment by nadgeri14** [comment]
@brendan-ai2 @matt-gardner Hey sorry for lately being inactive, I have exams, as a result, I could not attend it. I will try to close the PR by the end of next week.

**Comment by schmmd** [comment]
@nadgeri14 no worries--thanks for the contribution!

**Comment by schmmd** [comment]
@nadgeri14 should we leave this open?  Do you still expect to get to this?

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Allow the order of examples in the task cards to be specified explicitly
 - `histogram_interval` parameter is now deprecated in `TensorboardWriter`, please use `distribution_interval` instead.
 - Memory usage is not logged in tensorboard during training now. `ConsoleLoggerCallback` should be used instead.
+- If you use the `min_count` parameter of the Vocabulary, but you specify a namespace that does not exist, the vocabulary creation should raise a `ConfigurationError`.
```
Comment (dirkgr) [inline CHANGELOG.md:15]: Just change the wording here. It no longer "should". Now it "does"!

**Review by dirkgr — APPROVED** [review]
Just a minor comment on the changelog. Feel free to merge!

**Review by dirkgr — DISMISSED** [review]

**Review by AkshitaB — APPROVED** [review]

**Pull request merged**